crypto/keys/hd/hdpath.go

```diff
@@ -91,15 +79,30 @@ func NewParamsFromPath(path string) (*BIP44Params, error) {
 	if err != nil {
 		return nil, err
 	}
-	if !(change == 0 || change == 1) {
-		return nil, fmt.Errorf("change field can only be 0 or 1")
-	}
 
 	addressIdx, err := hardenedInt(spl[4])
 	if err != nil {
 		return nil, err
 	}
 
+	// Confirm valid values
```

Comment:
Why did this get moved down?

Reply:
Because the order to the check did not make sense. It was checking that values could be parsed after they were being used.

For instance, before things like this were happening. Where Foo/Bar/0/0/0 was returning something like "second and third... must he hardened" instead of saying that it was not even possible to parse the numbers.
```
	if !isHardened(spl[1]) || !isHardened(spl[2]) {
		return nil,
			fmt.Errorf("second and third field in path must be hardened (ie. contain the suffix ', got %v and %v", spl[1], spl[2])
	}

	coinType, err := hardenedInt(spl[1])
	if err != nil {
		return nil, err
	}

```

Actually, code like this could be simplified:
```
	// Confirm valid values
	if spl[0] != "44'" {
		return nil, fmt.Errorf("first field in path must be 44', got %v", spl[0])
	}

	// Check items can be parsed
	purpose, err := hardenedInt(spl[0])
	if err != nil {
		return nil, err
	}
```

Actually, this example for instance, could be simplified by:
```
	// Confirm valid values
	if spl[0] != "44'" {
		return nil, fmt.Errorf("first field in path must be 44', got %v", spl[0])
	}
        purpose = 44
```

Maybe we can add another issue to improve all this section? Otherwise, these changes seem a bit of of the scope of the ticket. I can revert this small change if necessary.

Reply:
🙈 Thank you for fixing this. I'm fine having those improvements in this PR.

Reply:
Also thank you for the explanation.